lib/shopping_cart.rb

```diff
@@ -0,0 +1,73 @@
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @full = false
```

Comment:
I don't hate the idea of an instance variable for `@full` here but it also seems we like don't need it. We should only have instance variables for values we intend to change, and are not dynamically calculated. This instance variable does not meet either of those criteria as it only being used once one line 34.